internal/datastore/crdb/reader.go

```diff
@@ -112,6 +112,37 @@ func (cr *crdbReader) ListNamespaces(ctx context.Context) ([]*core.NamespaceDefi
 	return nsDefs, nil
 }
 
+func (cr *crdbReader) LookupNamespaces(ctx context.Context, nsNames []string) ([]*core.NamespaceDefinition, error) {
+	if len(nsNames) == 0 {
+		return nil, nil
+	}
+
+	ctx = datastore.SeparateContextWithTracing(ctx)
+
+	var nsDefs []*core.NamespaceDefinition
+	if err := cr.execute(ctx, func(ctx context.Context) error {
+		tx, txCleanup, err := cr.txSource(ctx)
+		if err != nil {
+			return err
+		}
+		defer txCleanup(ctx)
+
+		nsDefs, err = lookupNamespaces(ctx, tx, nsNames)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf(errUnableToListNamespaces, err)
+	}
+
+	for _, nsDef := range nsDefs {
+		cr.addOverlapKey(nsDef.Name)
+	}
```

Comment:
for very large namespaces (presumably the ones causing the flakiness we are attempting to fix in this PR) this would lead to _many_ "touches" in the transactions table:

https://github.com/authzed/spicedb/blob/ffdf5210c7a97ae50b3f3d06070c04c4e7ea9e15/internal/datastore/crdb/crdb.go#L260-L264

@ecordell is there are reason we don't batch all those query operations instead of looping over the keyset?

Reply:
No real reason other than almost all requests should have only 1 overlap key

Reply:
Awesome, I think it would be relatively easy to turn that into a batch pgx request